Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionLibWPE.cpp

```diff
@@ -176,6 +269,50 @@ static void doKeyStrokeEvent(struct wpe_view_backend* viewBackend, bool pressed,
     }
 }
 
+static void doKeyStrokeEvent(WebPageProxy &page, bool pressed, uint32_t keyCode, uint32_t modifiers, bool doReleaseAfterPress = false)
+{
+#if ENABLE(WPE_PLATFORM)
+    if (auto* view = page.wpeView()) {
+
+        auto* display = wpe_display_get_default();
+        if (!display) {
+            LOG(Automation, "WebAutomationSession::doKeyStrokeEvent: Failed to find WPE display. Ignoring event.");
+            return;
+        }
+        GError* error = nullptr;
+        // FIXME what if it isn't xkb?
+        auto* keymap = WPE_KEYMAP_XKB(wpe_display_get_keymap(display, &error));
+        if (error) {
+            LOG(Automation, "WebAutomationSession::doKeyStrokeEvent: Failed to get keymap. Ignoring event.");
+            g_error_free(error);
+            return;
+        }
+
+        auto* xkbState = wpe_keymap_xkb_get_xkb_state(keymap);
+        unsigned keyval = xkb_state_key_get_one_sym(xkbState, keyCode);
+        if (keyval == XKB_KEY_NoSymbol)
+            return;
```

Comment:
We should add wpe_keymap_get_entries_for_keycode() to the new API, so that you don't need to deal with xkb

Reply:
Sorry, I meant wpe_keymap_translate_keyboard_state or similar.

Reply:
I removed the xkb specific code by using `wpe_keymap_get_entries_for_keyval`, the new API equivalent of `wpe_input_xkb_context_get_entries_for_key_code`.